numpy/matrixlib/tests/test_matrix_linalg.py

```diff
@@ -0,0 +1,96 @@
+""" Test functions for linalg module using the matrix class."""
+from __future__ import division, absolute_import, print_function
+
+import numpy as np
+
+from numpy.linalg.tests.test_linalg import (
+    LinalgCase, apply_tag, _TestNorm2D, TestQR as _TestQR, LinalgTestCase,
+    SolveCases, InvCases, EigvalsCases, EigCases, SVDCases, CondCases,
+    PinvCases, DetCases, LstsqCases)
+
+
+CASES = []
+
+# square test cases
+CASES += apply_tag('square', [
+    LinalgCase("0x0_matrix",
+               np.empty((0, 0), dtype=np.double).view(np.matrix),
+               np.empty((0, 1), dtype=np.double).view(np.matrix),
+               tags={'size-0'}),
+    LinalgCase("matrix_b_only",
+               np.array([[1., 2.], [3., 4.]]),
+               np.matrix([2., 1.]).T),
+    LinalgCase("matrix_a_and_b",
+               np.matrix([[1., 2.], [3., 4.]]),
+               np.matrix([2., 1.]).T),
+])
+
+# hermitian test-cases
+CASES += apply_tag('hermitian', [
+    LinalgCase("hmatrix_a_and_b",
+               np.matrix([[1., 2.], [2., 1.]]),
+               None),
+])
+# No need to make generalized or strided cases for matrices.
+
+
+class MatrixTestCase(LinalgTestCase):
+    TEST_CASES = CASES
+
+
+class TestSolve(SolveCases, MatrixTestCase):
+    pass
+
+
+class TestInv(InvCases, MatrixTestCase):
+    pass
+
+
+class TestEigvals(EigvalsCases, MatrixTestCase):
+    pass
+
+
+class TestEig(EigCases, MatrixTestCase):
+    pass
+
+
+class TestSVD(SVDCases, MatrixTestCase):
+    pass
+
+
+class TestCond(CondCases, MatrixTestCase):
+    pass
+
+
+class TestPinv(PinvCases, MatrixTestCase):
+    pass
+
+
+class TestDet(DetCases, MatrixTestCase):
+    pass
+
+
+class TestLstsq(LstsqCases, MatrixTestCase):
+    pass
+
+
+class TestNormDouble(_TestNorm2D):
+    array = np.matrix
+    dt = np.double
+    dec = 12
+
+
+class TestNormSingle(_TestNorm2D):
+    array = np.matrix
+    dt = np.float32
+    dec = 6
+
+
+class TestNormInt64(_TestNorm2D):
+    array = np.matrix
+    dt = np.int64
+    dec = 12
```

Comment:
To me. it's tempting to define in the main linalg tests

```python
# these base classes are extracted for use by the matrix tests
class _TestNormDoubleBase(_TestNormBase):
   dt = np.double
   dec = 12

class _TestNormSingleBase(_TestNormBase):
   dt = np.single
   dec = 6

class _TestNormInt64Base(_TestNormBase):
   dt = np.int64
   dec = 12

class TestNormDouble(_TestNormDoubleBase, _TestNorm): pass
class TestNormSingle(_TestNormSingleBase, _TestNorm): pass
class TestNormInt64(_TestNormInt64Base, _TestNorm): pass
```

And then in matrixlib, you can do

```python
class _TestNorm2DMatrix(linalg._TestNorm2D):
    array = np.matrix

class TestNormDouble(linalg._TestNormDoubleBase, _TestNorm2DMatrix): pass
class TestNormSingle(linalg._TestNormSingleBase, _TestNorm2DMatrix): pass
class TestNormInt64(linalg._TestNormInt64Base, _TestNorm2DMatrix): pass
```

Reply:
OK, done, and also called all the test classes `...Matrix`